eclipse-theia/theia
# theming: Properly Apply Default Icon Theme

#### What it does
Closes #11837 
The issue was caused by the editor reading the theme before the active theme was changed.
In other words, the editor read the previous theme and placed the classes for those icons. Once the theme changed those class icons had no icons associated with them and as such were not able to load.

This commit solves the issue by changing the order of instructions. Assigning the new theme to the active parameter before moving on to the disposing and activating the theme.

#### How to test
1. Run Theia
2. Open the Icon Theme menu quick pick
3. Use the arrow keys to highlight/select between themes
4. Check that all icon themes display properly

Alternatively, you can use the mouse or arrow keys to select and change between themes.

![Issue11837Test01](https://user-images.githubusercontent.com/48699277/232066456-1b4c1231-31a6-4a45-bcf5-6d1176f4b186.gif)

##### Notes
Currently, disposing of the current theme triggers the editor to set labels and thus read the icon themes.
The plugins add another event trigger. The `none` theme has this event trigger twice. One in the middle of setting the new theme (the method `setCurrent` which is default behavior) then another one after the method finishes. Probably due to being tied with the `IconThemeService`. The Default/Theia theme only triggered this "reload" event once and thus was the one where the issue was noticeable. This double trigger in the `none` theme does not happen when plugins are not installed.

#### Review checklist


#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
